openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -338,3 +348,58 @@ def get(self, request, course_key_string, *args, **kwargs):
             pass
 
         return Response(resp)
+
+
+class Celebration(DeveloperErrorViewMixin, APIView):
+    """
+    **Use Cases**
+
+        Marks a particular celebration as complete
+
+    **Example Requests**
+
+        POST /api/courseware/celebration/{course_key}
+
+    **Request Parameters**
+
+        Body consists of the following fields:
+
+            * first_section (bool): when a user finishes their first section of a course
+
+    **Returns**
+
+        * 200 on success with above fields.
+        * 400 if an invalid parameter was sent.
+        * 404 if the course is not available or cannot be seen.
+    """
+
+    authentication_classes = (
+        JwtAuthentication,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (IsAuthenticated, )
+    http_method_names = ['post']
```

Comment:
Is this necessary? I don't remember seeing it in other APIs

Reply:
Yeah it was necessary, but I'm not super familiar with why. Seems like we are using a particular DRF class here that wanted that specified to allow a post call.

I'm more familiar with the DRF pattern of `create` calls and mixins that map POST to create. But the classes in this file weren't using that pattern. I dunno.

Reply:
Yeah, same. Sounds good.